collate.go

```diff
@@ -56,8 +56,15 @@ func (c *JSONCollator) Collate(key1, key2 interface{}) int {
 	case kString:
 		return c.compareStrings(key1.(string), key2.(string))
 	case kArray:
-		array1 := key1.([]interface{})
-		array2 := key2.([]interface{})
+		// Handle the case where a walrus bucket is returning a []float64
+		array1, ok1 := key1.([]interface{})
+		if !ok1 {
+			array1 = ToArrayOfInterface(key1.([]float64))
```

Comment:
This code is a bit non-intuitive and potentially brittle - it makes the assumption that converting key1 to []float64 will succeed, based on knowledge of the type switch in collationType().

I think it would be clearer/more readable to just type switch on key1 and key2 here (and return 0 in the case it's not []interface{} or []float64).